### PR TITLE
Optional fields which were mandatory earlier creating failure with older service version

### PR DESCRIPTION
**Pull request by tanyasethi-msft**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [x] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [x] Tests are included and/or updated for code changes.
- [ ] Updates to module CHANGELOG.md are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go

### CODE REVIEW

**Comment by RickWinter** [comment]
Please add an entry to CHANGELOG.md as part of this fix.

**Review by vibhansa-msft — APPROVED** [review]

File: sdk/storage/azfile/internal/exported/smb_property.go [hunk 1]
```diff
@@ -34,12 +35,12 @@ func (sp *SMBProperties) Format(isDir bool, defaultFileAttributes string, defaul
 }
 
 // FormatSMBProperties returns file attributes, creation time, last write time and change time.
-func FormatSMBProperties(sp *SMBProperties, isDir bool) (fileAttributes *string, creationTime *string, lastWriteTime *string, changeTime *string) {
+func FormatSMBProperties(sp *SMBProperties, defaultAttributes *string, defaultCurrentTime *string, isDir bool) (fileAttributes *string, creationTime *string, lastWriteTime *string, changeTime *string) {
 	if sp == nil {
-		return nil, nil, nil, nil
+		return defaultAttributes, defaultCurrentTime, defaultCurrentTime, to.Ptr(shared.DefaultCurrentTimeString)
```
Comment (souravgupta-msft) [inline sdk/storage/azfile/internal/exported/smb_property.go:40]: Return `nil` for `changeTime`.

**Review by souravgupta-msft — APPROVED** [review]

File: sdk/storage/azfile/internal/exported/smb_property.go (hunk 1 above)
Comment (tanyasethi-msft) [reply sdk/storage/azfile/internal/exported/smb_property.go:40]: Changed

File: sdk/storage/azfile/directory/models.go [hunk 2]
```diff
@@ -39,12 +41,17 @@ type CreateOptions struct {
...
+			FileLastWriteTime: to.Ptr(shared.DefaultCurrentTimeString),
+			FilePermission:    to.Ptr(shared.DefaultFilePermissionString),
+		}
 	}
 
-	fileAttributes, fileCreationTime, fileLastWriteTime, fileChangeTime := exported.FormatSMBProperties(o.FileSMBProperties, true)
+	fileAttributes, fileCreationTime, fileLastWriteTime, fileChangeTime := exported.FormatSMBProperties(o.FileSMBProperties, to.Ptr(shared.FileAttributesDirectory), to.Ptr(shared.DefaultCurrentTimeString), true)
 
-	permission, permissionKey := exported.FormatPermissions(o.FilePermissions)
+	permission, permissionKey := exported.FormatPermissions(o.FilePermissions, to.Ptr(shared.DefaultFilePermissionString))
```
Comment (vibhansa-msft) [inline sdk/storage/azfile/directory/models.go:54]: Update CHANGELOG.md as part of this fix.

**Pull request merged**